crates/ra_project_model/src/lib.rs

```diff
@@ -200,9 +205,17 @@ impl ProjectWorkspace {
     /// the root is a member of the current workspace
     pub fn to_roots(&self) -> Vec<PackageRoot> {
         match self {
-            ProjectWorkspace::Json { project } => {
-                project.roots.iter().map(|r| PackageRoot::new_member(r.path.clone())).collect()
-            }
+            ProjectWorkspace::Json { project, project_location } => project
+                .roots
+                .iter()
+                .map(|r| {
+                    let root_path = match r.path.is_absolute() {
+                        true => r.path.clone(),
+                        false => project_location.join(&r.path),
+                    };
```

Comment:
```suggestion
                    let root_path = project_location.join(&r.path);
```

`join` already replaces in case of absolute paths.

https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#examples-4

> Pushing an absolute path replaces the existing path:

Reply:
Thanks. Done.